### PR TITLE
CO-68 Expand datamodel to represent truck fueltank state.

### PR DESCRIPTION
**Pull request by JeppeTrip**
Problems with the database not updating as it should (which is why I also added the dbscript into this commit).

### CODE REVIEW

File: backend/Domain/Entities/TruckState.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+using Domain.Common;
+
+namespace Domain.Entities
+{
+  public class TruckState : AuditableEntity
+  {
+    public int Id { get; set; }
+    public double MorningQuantity { get; set; }
+    public double EveningQuantity { get; set; }
+    public ICollection<TruckRefill> Refills { get; set; }
```
Comment (objmj-itminds) [inline backend/Domain/Entities/TruckState.cs:11]: Let's reverse this and put a Truck and TruckId into the TruckRefill entity

File: backend/Domain/Entities/Truck.cs [hunk 2]
```diff
@@ -6,6 +7,7 @@ public class Truck : AuditableEntity
   {
     public int Id { get; set; }
     public Route Route { get; set; }
+    public ICollection<TruckState> DailyStates { get; set; }
```
Comment (objmj-itminds) [inline backend/Domain/Entities/Truck.cs:10]: Let's reverse this and put a Truck and TruckId into the TruckState entity.

We can always keep these as `virtual`

**Review by objmj-itminds — CHANGES_REQUESTED** [review]
Maybe the cause due to an empty migration. I can help you look at this isuee.

File: backend/Domain/Entities/TruckState.cs (hunk 1 above)
Comment (JeppeTrip) [reply backend/Domain/Entities/TruckState.cs:11]: You mean put the TruckState and TruckStateId into TruckRefill entity for this one, right?

File: backend/Infrastructure/Persistence/Migrations/20210113092515_TruckStateAndRefillsModel.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
```
Comment (CSCITM) [inline backend/Infrastructure/Persistence/Migrations/20210113092515_TruckStateAndRefillsModel.cs:1]: Remove empty migration

File: backend/Domain/Entities/TruckState.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+namespace Domain.Entities
+{
+  public class TruckState : AuditableEntity
+  {
+    public int Id { get; set; }
+    public double MorningQuantity { get; set; }
+    public double EveningQuantity { get; set; }
+    public int TruckId { get; set; }
+    public Truck Truck { get; set; }
+    public virtual ICollection<TruckRefill> Refills { get; set; }
```
Comment (CSCITM) [inline backend/Domain/Entities/TruckState.cs:13]: I see virtual in this PR but not in the other, any reason?

File: backend/Web/dbscript [hunk 5]
```diff
@@ -0,0 +1,172 @@
+﻿CREATE TABLE [ExampleEntityLists] (
```
Comment (CSCITM) [inline backend/Web/dbscript:1]: What is the use of this?

**Comment by objmj-itminds** [comment]
I think the best approach is to sorta "start from scratch" with this. A whole new branch from main and then restart with the entities and migrations.

File: backend/Domain/Entities/TruckState.cs (hunk 1 above)
Comment (objmj-itminds) [reply backend/Domain/Entities/TruckState.cs]: Yes

**Pull request closed without merging**